test/Conversion/FIRRTLToHW/lower-to-hw.mlir

```diff
@@ -1285,4 +1285,191 @@ firrtl.circuit "Simple"   attributes {annotations = [{class =
     firrtl.connect %o3, %r3 : !firrtl.uint<32>, !firrtl.uint<32>
     firrtl.connect %o4, %r4 : !firrtl.uint<100>, !firrtl.uint<100>
   }
+  // CHECK-LABEL: hw.module @init1DVector
+  firrtl.module @init1DVector(in %clock: !firrtl.clock, in %a: !firrtl.vector<uint<1>, 2>, out %b: !firrtl.vector<uint<1>, 2>) {
+    %r = firrtl.reg %clock  : !firrtl.vector<uint<1>, 2>
+    // CHECK:      sv.ifdef.procedural "RANDOMIZE_REG_INIT"  {
+    // CHECK-NEXT:   %1 = sv.array_index_inout %r[%false] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %2 = comb.extract %RANDOM from 0 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %1, %2 : i1
+    // CHECK-NEXT:   %3 = sv.array_index_inout %r[%true] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %4 = comb.extract %RANDOM from 1 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %3, %4 : i1
+    // CHECK-NEXT: }
+
+    firrtl.connect %r, %a : !firrtl.vector<uint<1>, 2>, !firrtl.vector<uint<1>, 2>
+    firrtl.connect %b, %r : !firrtl.vector<uint<1>, 2>, !firrtl.vector<uint<1>, 2>
+    // CHECK:      sv.always posedge %clock  {
+    // CHECK-NEXT:   sv.passign %r, %a : !hw.array<2xi1>
+    // CHECK-NEXT: }
+    // CHECK-NEXT: hw.output %0 : !hw.array<2xi1>
+  }
+
+  // CHECK-LABEL: hw.module @init2DVector
+  firrtl.module @init2DVector(in %clock: !firrtl.clock, in %a: !firrtl.vector<vector<uint<1>, 2>, 2>, out %b: !firrtl.vector<vector<uint<1>, 2>, 2>) {
+    %r = firrtl.reg %clock  : !firrtl.vector<vector<uint<1>, 2>, 2>
+    // CHECK:      sv.ifdef.procedural "RANDOMIZE_REG_INIT"  {
+    // CHECK-NEXT:   %1 = sv.array_index_inout %r[%false] : !hw.inout<array<2xarray<2xi1>>>, i1
+    // CHECK-NEXT:   %2 = sv.array_index_inout %1[%false] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %RANDOM = sv.verbatim.expr.se "`RANDOM" : () -> i32 {symbols = []}
+    // CHECK-NEXT:   %3 = comb.extract %RANDOM from 0 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %2, %3 : i1
+    // CHECK-NEXT:   %4 = sv.array_index_inout %1[%true] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %5 = comb.extract %RANDOM from 1 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %4, %5 : i1
+    // CHECK-NEXT:   %6 = sv.array_index_inout %r[%true] : !hw.inout<array<2xarray<2xi1>>>, i1
+    // CHECK-NEXT:   %7 = sv.array_index_inout %6[%false] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %8 = comb.extract %RANDOM from 2 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %7, %8 : i1
+    // CHECK-NEXT:   %9 = sv.array_index_inout %6[%true] : !hw.inout<array<2xi1>>, i1
+    // CHECK-NEXT:   %10 = comb.extract %RANDOM from 3 : (i32) -> i1
+    // CHECK-NEXT:   sv.bpassign %9, %10 : i1
+    // CHECK-NEXT: }
+    firrtl.connect %r, %a : !firrtl.vector<vector<uint<1>, 2>, 2>, !firrtl.vector<vector<uint<1>, 2>, 2>
+    firrtl.connect %b, %r : !firrtl.vector<vector<uint<1>, 2>, 2>, !firrtl.vector<vector<uint<1>, 2>, 2>
```

Comment:
This and the other tests here are quite large. This usually makes me nervous from a maintainability perspective. Are there any ways to reduce the size of the test or limit what is checked. One idea is to use `firrtl.vector<vector<uint<1>, 1>, 1>`. That said, we do need to test the non-corner case of `vector<_, 2>`. 🤔